.github/workflows/build_pcluster_amis.yml

```
@@ -0,0 +1,97 @@
+name: Build pcluster AMIs
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+# Cancel any workflows that are already running
+concurrency:
+  group: ${{ github.workflow }}
+  cancel-in-progress: true
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  build:
+    name: Build AMI [${{ matrix.architecture }}, ${{ matrix.aws-region }}]
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - architecture: x86_64
+            source-ami-id: ami-011244318c573c915
+            aws-region: us-east-1
+          - architecture: arm64
+            source-ami-id: ami-0d87b43677ef96ab9
+            aws-region: us-east-1
+          - architecture: x86_64
+            source-ami-id: ami-08f1611028f2254b3
+            aws-region: us-west-2
+          - architecture: arm64
+            source-ami-id: ami-0597a0565f343efb8
+            aws-region: us-west-2
+    steps:
+      - name: Checkout `spack-infrastructure` repo
+        uses: actions/checkout@v3
+
+      - name: Checkout `awslabs/amazon-eks-ami` repo
+        uses: actions/checkout@v3
+        with:
+          repository: awslabs/amazon-eks-ami
+          path: amazon-eks-ami
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v2.0.0
+        with:
+          role-to-assume: arn:aws:iam::588562868276:role/GitHubActionsRole
+          aws-region: ${{ matrix.aws-region }}
+
+      - name: Setup packer
+        uses: hashicorp/setup-packer@v2.0.0
+        with:
+          version: latest
+
+      - name: Setup terraform
+        uses: hashicorp/setup-terraform@v2
+        with:
+          # Disable the github actions "wrapper" for Terraform
+          # because it transforms the output such that it can't
+          # be read by `jq`.
+          terraform_wrapper: false
+
+      - name: Initialize terraform
+        working-directory: terraform/production
+        run: terraform init
```

```suggestion
        run: terraform init -input=false
```
https://developer.hashicorp.com/terraform/tutorials/automation/automate-terraform#automated-terraform-cli-workflow